drivers/video/ov5640.c

```diff
@@ -989,17 +1225,31 @@ static int ov5640_init(const struct device *dev)
 	k_sleep(K_MSEC(5));
 
 	/* Initialize register values */
-	ret = ov5640_write_multi_regs(&cfg->i2c, init_params, ARRAY_SIZE(init_params));
+	ret = ov5640_write_multi_regs(&cfg->i2c, init_params_common,
+				      ARRAY_SIZE(init_params_common));
 	if (ret) {
 		LOG_ERR("Unable to initialize the sensor");
 		return -EIO;
 	}
 
+	if (ov5640_is_dvp(dev)) {
+		ret = ov5640_write_multi_regs(&cfg->i2c, init_params_dvp,
+					      ARRAY_SIZE(init_params_dvp));
+
+		if (ret) {
+			LOG_ERR("Unable to initialize the sensor with DVP parameters");
+			return -EIO;
+		}
+	}
+
 	/* Set virtual channel */
-	ret = ov5640_modify_reg(&cfg->i2c, 0x4814, 3U << 6, (uint8_t)(DEFAULT_MIPI_CHANNEL) << 6);
-	if (ret) {
-		LOG_ERR("Unable to set virtual channel");
-		return -EIO;
+	if (!ov5640_is_dvp(dev)) {
+		ret = ov5640_modify_reg(&cfg->i2c, 0x4814, 3U << 6,
+					(uint8_t)(DEFAULT_MIPI_CHANNEL) << 6);
```

Comment:
Absolutely non-blocking, minor optimization: create a `init_params_csi2` table where all MIPI/CSI2 parameters can be added.
```suggestion
} else {
		ret = ov5640_write_multi_regs(&cfg->i2c, init_params_csi2,
					      ARRAY_SIZE(init_params_csi2));
		if (ret) {
			LOG_ERR("Unable to set virtual channel");
			return -EIO;
		}
```

This is functionally equivalent to what you did though!

Reply:
The MIPI/CSI2 parameters are already into the `init_params_common`. Maybe in the future could be interesting to check those parameters carefully to reduce unnecessary parameter configuration, as you mentioned below.